### PR TITLE
Add support for custom error message; test assertion failure

### PR DESCRIPTION
**Pull request by forshtat**

### CODE REVIEW

**Comment by nventuro** [comment]
Thanks for opening this @forshtat! I personally don't often use custom chai messages, but I can see how they could be useful. 

Regarding the PR itself, I'd like to keep the test scenarios where the custom error messages are not supplied. What do you think about adding a `message` argument to `testerGenerator`, and then calling 
 * `expect(a).to.be.a.bignumber.that[functionName](b, message)` if there is a `message`, or 
 * `expect(a).to.be.a.bignumber.that[functionName](b)` otherwise?

We could alternatively add the new argument to the `tester, notTester` functions, which might make the tests easier to read. Feel free to refactor those as you see fit!

**Comment by forshtat** [comment]
@nventuro do you meane something like this: https://github.com/OpenZeppelin/chai-bn/pull/9/commits/4afc1f82bf8942f73f092f19a4e3c92c6cb22542?

**Comment by nventuro** [comment]
Yes, thanks a lot! 

I'd suggest doing the following small changes to the tests before we get this merged:
 1. To avoid coupling our tests too tight to Chai's error messages, I wouldn't perform any check on default errors (just call `.should.throw()`)
 1. I'd make the custom error message check reusable by removing the bits about `.equal`: `Custom message: *`
 1. Can you add test cases for all other operations? `above`, `least`, `closeTo`, etc.

Thanks again!

**Comment by nventuro** [comment]
Sorry, I closed this by mistake!

**Comment by forshtat** [comment]
@nventuro 

1. Done. In general, I would suggest you consider coupling your tests with the underlying error messages. If you don't do that, you cannot be sure the expected exception was indeed thrown or there was some null pointer.
2. Done. Then again, the actual underlying error is not tested.
3. Done.

File: test/chai-bn.test.js [hunk 1]
```diff
@@ -126,232 +124,299 @@ describe('chai-bn', function () {
...
-
-      testCases.forEach(([a, b]) => {
+      notAtLeastTestCases.forEach(([a, b]) => {
         notTester(a, b);
         a.should.not.be.a.bignumber.that.is.at.least(b);
         expect(a).to.not.be.a.bignumber.that.is.at.least(b);
       });
     });
 
+    it('above fails on unaboveness', function () {
```
Comment (nventuro) [inline test/chai-bn.test.js]:
```suggestion
    it('at least fails fails on unaboveness', function () {
```

File: test/chai-bn.test.js [hunk 2]
```diff
@@ -126,232 +124,299 @@ describe('chai-bn', function () {
...
+      notAtLeastTestCases.forEach(([a, b]) => {
+        (() => tester(a, b)).should.throw();
+        (() => tester(a, b, customMessage)).should.throw(customMessageRegex);
+      });
+    });
+
+    it('not above fails on aboveness', function () {
```
Comment (nventuro) [inline test/chai-bn.test.js]:
```suggestion
    it('at most fails on aboveness', function () {
```

**Review by nventuro — APPROVED** [review]
Looks great, thanks a lot @forshtat!

**Pull request merged**